python/cudf/utils/cudautils.py

```diff
@@ -580,6 +580,24 @@ def gpu_unique_set_insert(vset, sz, val):
     return -1
 
 
+@cuda.jit
+def gpu_shift(in_col, out_col, N):
+    """Shift value at index i of an input array forward by N positions and
+    store the output in a new array.
+    """
+    i = cuda.grid(1)
+    if N > 0:
+        if i < in_col.size:
+            out_col[i] = in_col[i-N]
+        if i < N:
+            out_col[i] = -1
```

Comment:
If we handle nulls, we can combine these two statements to just look like:
```
if i >=N and i < in_col.size:
```
that way we don't unnecessarily set the data where the nullmask bit would be set anyway.